spec/requests/comment_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Comments', type: :request do
+  describe 'NEW post comment' do
```

Comment:
I would define the whole describe as the 'POST #create'.

Inside, there would be two different contexts, one for the comment of a post and another for the comment of a comment.

By doing this you avoid defining similar variables like you're doing right now.